app/services/listing_service/store/shape.rb

```diff
@@ -51,22 +62,67 @@ def create(community_id:, opts:)
     shape = NewShape.call(opts.merge(community_id: community_id))
 
     units = shape[:units].map { |unit| Unit.call(unit) }
+    raise NotImplementedError.new("For backward compatibility reasons saving multiple units is not yet supported") if units.length > 1
+
     translations = opts[:translations] # Skip data type and validation, because this is temporary
 
-    # TODO We should be able to create transaction_type without community
-    community = Community.find(shape[:community_id])
+    tt_model = nil
+
+    ActiveRecord::Base.transaction do
+      # TODO We should be able to create transaction_type without community
+      community = Community.find(shape[:community_id])
+
+      create_tt_opts = to_tt_model_attributes(shape).except(:units, :translations)
+      tt_model = community.transaction_types.build(create_tt_opts)
+
+      units.each { |unit|
+        tt_model.listing_units.build(to_unit_model_attributes(unit))
+      }
+      translations.each { |tr| tt_model.translations.build(tr) }
+
+      tt_model.save!
+    end
+
+    from_transaction_type_model(tt_model)
+  end
 
-    create_tt_opts = to_tt_model_attributes(shape).except(:units, :translations)
-    tt_model = community.transaction_types.build(create_tt_opts)
+  def update(community_id:, transaction_type_id: nil, listing_shape_id: nil, opts:)
+    tt_model = if transaction_type_id
+      TransactionTypeModel.where(community_id: community_id, id: transaction_type_id).first
+    elsif listing_shape_id
+      raise NotImplementedError.new("Can not find listing shape by listing_shape_id, yet. Specify transaction_type_id instead.")
+    else
+      raise ArgumentError.new("Can not find listing shape without id.")
+    end
+
+    return nil if tt_model.nil?
 
+    update_shape = UpdateShape.call(opts.merge(community_id: community_id))
+
+    units = update_shape[:units].map { |unit| Unit.call(unit) }
     raise NotImplementedError.new("For backward compatibility reasons saving multiple units is not yet supported") if units.length > 1
 
-    units.each { |unit|
-      tt_model.listing_units.build(to_unit_model_attributes(unit))
-    }
-    translations.each { |tr| tt_model.translations.build(tr) }
+    translations = opts[:translations] # Skip data type and validation, because this is temporary
+
+    # TODO We should be able to create transaction_type without community
+    ActiveRecord::Base.transaction do
+      community = Community.find(community_id)
+
+      update_tt_opts = HashUtils.compact(to_tt_model_attributes(update_shape)).except(:units, :translations)
+      tt_model.update_attributes(update_tt_opts)
+
+      unless units.nil?
+        tt_model.listing_units.destroy_all
+        units.each { |unit| tt_model.listing_units.build(to_unit_model_attributes(unit)) }
+      end
+
+      unless translations.nil?
```

Comment:
So update rebuilds all unit rows every time? This implies that units cannot ever be referenced from outside.